### DISCUSSION TITLE
How should the middleware matcher look like if the default locale is ommited?

### DISCUSSION THREAD

**Discussion by alexbuznik-stark** [amannn/next-intl#1624] [Q&A] [answered]
Hi.

The default middleware matcher from the tutorial is `matcher: ['/', '/(de|en)/:path*']`

I want to support a web-site that has default locale ommited (en-US).

How should the matcher look like then?
Are the matchers applied in order?

My suggestion, but I'm still not sure it works well for all cases.
```
  matcher: [
    "/",
    `/(${locales.join("|")})/:path*`,
    "/:path*",
    '/((?!api|_next/static|_next/image|assets|favicon.ico|sw.js|site.webmanifest).*)',
  ],
```

**Comment by amannn** [comment]
You mean you're using `localePrefix: 'as-needed'`? This is the one that's suggested in the docs: [Pathnames without a locale prefix](https://next-intl.dev/docs/routing/middleware#matcher-no-prefix)

**Reply by alexbuznik-stark** [reply]
Yes, thank you.

But I have many pathnames, and I want to apply i18n to all of them except static files, so is there a way to do it without listing all of my pathnames?

**Reply by amannn** [reply]
The example in the docs doesn't list all pathnames, right? It's a negative lookup that ignores certain paths, while explicitly mentioning those where a dot is expected.

Can you provide some examples?

**Reply by alexbuznik-stark** [reply]
Well, I'm a bit a confused with examples.

So the [tutorial](https://next-intl.dev/docs/getting-started/app-router/with-i18n-routing#middleware) has this matcher:
```
  matcher: ['/', '/(de|en)/:path*']
```

Then, the [example from the repo](https://github.com/amannn/next-intl/blob/main/examples/example-app-router/src/middleware.ts) has this which lists root and paths localized:
```
  matcher: [
    // Enable a redirect to a matching locale at the root
    '/',

    // Set a cookie to remember the previous locale for
    // all requests that have a locale prefix
    '/(de|en)/:path*',

    // Enable redirects that add missing locales
    // (e.g. `/pathnames` -> `/en/pathnames`)
    '/((?!_next|_vercel|.*\\..*).*)'
  ]
```

**Reply by alexbuznik-stark** [reply]
So the question is it going to suffice the matcher with the negative lookup for the static assets to make it work on all paths with `localePrefix: 'as-needed'`:

```
    matcher: ['/((?!_next|_vercel|.*\\..*).*)']
```

**Reply by amannn** [reply, marked as answer]
It depends on your app, have you read the section [Pathnames without a locale prefix](https://next-intl.dev/docs/routing/middleware#matcher-no-prefix)?

**Reply by alexbuznik-stark** [reply]
Thank you, I have seen it.
